controllers/common.go

```diff
@@ -51,18 +51,17 @@ func setPodTemplateSpecDefaults(template *corev1.PodTemplateSpec) {
 	if template.Spec.SchedulerName == "" {
 		template.Spec.SchedulerName = "default-scheduler"
 	}
-	// for each volumes, if volumeSource is Projected or SecretVolumeSource, set default mode
+	// for each volumes, if volumeSource is Projected or SecretVolumeSource, set default permission
 	for i := range template.Spec.Volumes {
 		if template.Spec.Volumes[i].Projected != nil {
 			if template.Spec.Volumes[i].Projected != nil {
-				template.Spec.Volumes[i].Projected.DefaultMode = common.DefaultModePtr()
+				template.Spec.Volumes[i].Projected.DefaultMode = ptr.To(common.DefaultPermission)
 			}
 		} else if template.Spec.Volumes[i].Secret != nil {
```

Comment:
these `else if`s seem strange to me. Should not they be only `if`s?

Reply:
A given volume would only have one type. Using else if would probably save an unnecessary second if check.

Reply:
right, missed that
```go
// Represents the source of a volume to mount.
// Only one of its members may be specified.
type VolumeSource struct {
```